### PR TITLE
VPCI coding style and bugs fixes found in integration testing for partition mode

### PR DESCRIPTION
**Pull request by zds100**
Tracked-On: #1126
Signed-off-by: dongshen <dongsheng.x.zhang@intel.com>

### CODE REVIEW

File: hypervisor/partition/vm_description.c [hunk 1]
```diff
@@ -41,33 +41,43 @@ static struct vpci_vdev_array vpci_vdev_array1 = {
...
 		[5] = {
 		.base = 0UL,
-		.size = 0x2000UL,
+		.size = 0x1000UL,
 		.type = PCIBAR_MEM32
 		},
 	  },
 	 .pdev = {
-		.bdf = {.b = 0x00U, .d = 0x01U, .f = 0x00U},
+		.bdf = {.b = 0x00U, .d = 0x12U, .f= 0x00U},
```
Comment (anthonyzxu) [inline hypervisor/partition/vm_description.c]: need a space after .f before =

File: hypervisor/partition/vm_description.c [hunk 2]
```diff
@@ -76,43 +86,67 @@ static struct vpci_vdev_array vpci_vdev_array1 = {
 };
 
 static struct vpci_vdev_array vpci_vdev_array2 = {
-	.num_pci_vdev = 2,
+	.num_pci_vdev = 3,
 
 	.vpci_vdev_list = {
 	 {/*vdev 0: hostbridge*/
-	  .vbdf = {.b = 0x00U, .d = 0x00U, .f = 0x00U},
+	  .vbdf = { .b = 0x00U, .d = 0x00U, .f = 0x00U },
```
Comment (anthonyzxu) [inline hypervisor/partition/vm_description.c]: keep the style consistent,  remove space before .b, remove space before }

File: hypervisor/partition/vm_description.c [hunk 3]
```diff
@@ -76,43 +86,67 @@ static struct vpci_vdev_array vpci_vdev_array1 = {
...
 	  .ops = &pci_ops_vdev_hostbridge,
-	  .bar = {}, /* don't care for hostbridge */
-	  .pdev = {} /* don't care for hostbridge */
+	  .bar = {},
+	  .pdev = {
+		.bdf = { .b = 0x00U, .d = 0x00U, .f = 0x00U },
```
Comment (anthonyzxu) [inline hypervisor/partition/vm_description.c]: ditto

File: hypervisor/dm/vpci/pci_pt.c [hunk 4]
```diff
@@ -214,44 +219,47 @@ static int vdev_pt_remap_bar(struct pci_vdev *vdev, uint32_t idx,
...
-		error = vdev_pt_remap_bar(vdev, idx, pci_bar_base(new_bar));
-		if (error) {
-			pr_err("vdev_pt_remap_bar failed: %d", idx);
+	case PCIBAR_MEM32:
+		bar_update_normal = (new_bar_uos != (uint32_t)~0U);
+		new_bar = new_bar_uos & mask;
+		if (bar_update_normal) {
+			error = vdev_pt_remap_bar(vdev, idx,
+				pci_bar_base(new_bar));
+			if (error) {
```
Comment (anthonyzxu) [inline hypervisor/dm/vpci/pci_pt.c]: if (error != 0)

**Comment by zds100** [comment]
Recreated patches based on Anthony's comments.

File: hypervisor/partition/vm_description.c (hunk 1 above)
Comment (zds100) [reply hypervisor/partition/vm_description.c]: Addressed review comment.

File: hypervisor/partition/vm_description.c (hunk 2 above)
Comment (zds100) [reply hypervisor/partition/vm_description.c]: Addressed review comment.

**Comment by zds100** [comment]
Added "Tracken-On" to all commits

**Pull request merged**